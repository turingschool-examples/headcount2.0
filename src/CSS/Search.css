.search-bar {
  margin: 1rem 0.5rem;
}

.search-input {
  height: 2rem;
  width: 20rem;
  border: 1.5px solid #ccc;
  margin: 1rem 0;
  border-radius: 5px;
}

.search-input:hover {
  box-shadow: 1px 1px 1px 1px #ccc;
}

.hide-suggestions {
  display: none;
}

.did-you-mean {
  font-style: italic;
  color: gray;
}

.show-suggestions {
  margin: 1rem 0;
  text-align: center;
}

.suggestions {
  cursor: pointer;
  border: 1px solid gray;
  margin: 0;
  padding: 2px;
  text-align: center;
}

.suggestions:hover {
  background-color: rgb(244, 243, 243);
  box-shadow: 1.5px 1.5px 1.5px 1.5px #ccc;
}

button {
  width: 21.3rem;
  height: 2rem;
  border-radius: 5px;
  background-color: rgb(230, 230, 230);
  font-size: 1.1rem;
}

button:hover {
  background-color: rgb(142, 142, 142);
  color: rgb(237, 237, 237);
  cursor: pointer;
}

input[type="text"] {
  font-size: 1.1rem;
  line-height: 5rem;
  padding-left: 1rem;
}

.show {
  margin-bottom: 10rem;
  font-style: italic;
}

.show-none {
  display: none;
}

.display-error {
  padding: 1rem 0;
  text-align: center;
}
