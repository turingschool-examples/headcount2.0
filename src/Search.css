.search {
  text-align: center
}

.inputField {
  padding: 10px;
  border: 3px solid red;
  /*margin: 10px;*/
  width: 300px;
}

.inputButton {
  padding: 10px;
  background-color: white;
  margin: 10px;
  border-radius: 30%;
  border: 3px solid green;
  color: green;
}

.inputButton:hover {
  background-color: green;
  color: white;
}
