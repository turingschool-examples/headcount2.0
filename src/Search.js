import React, { Component } from "react";
import DistrictRepository from "./helper";
import kindergarners from "./data/kindergartners_in_full_day_program";

export default class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userInput: "",
      searchSuggestions: []
    };
    this.districtRepository = new DistrictRepository(kindergarners);
  }

  handleChange = ({ name, value }) => {
    this.setState({ [name]: value });
    this.suggestDistricts(this.state.userInput);
  };

  suggestDistricts = e => {
    let suggestions = this.districtRepository.findAllMatches(
      this.state.userInput
    );
    this.setState({ searchSuggestions: suggestions });
    if (e.length > 2) {
      this.props.handleSubmit(this.state.searchSuggestions);
    }
  };

  searchDistrict = district => {
    const filteredDistricts = this.districtRepository.findByName(district);
    this.props.handleSubmit([filteredDistricts]);
  };

  render() {
    return (
      <div>
        <form>
          <input
            placeholder="Search District"
            name="userInput"
            onChange={e => this.handleChange(e.target)}
            value={this.state.userInput}
          />
          {/* <button onClick={e => this.searchDistrict(e)}>Search</button> */}
        </form>
        <div>
          {this.state.searchSuggestions.map((district, i) => {
            if (
              this.state.searchSuggestions &&
              this.state.userInput.length > 3
            ) {
              return (
                <p
                  className="suggestions"
                  key={i}
                  onClick={e => this.searchDistrict(e.target.textContent)}
                >
                  {district.location}
                </p>
              );
            }
          })}
        </div>
      </div>
    );
  }
}
