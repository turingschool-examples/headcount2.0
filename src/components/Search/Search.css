/*SEARCH COMPONENT RELATED CSS GOES HERE*/
.Search {
  width: 35%;
  display: flex;
  justify-content: space-between
}

input {
  height: 2rem;
  width: 100%;
  border: 2px solid #BEBEBE;
  padding-left: .6rem;
  font-size: 1.25rem;
  letter-spacing: 0.5px;
  color: #006989;
}

::placeholder {
  font-family: "Roboto", sans-serif;
  font-weight: 400;
  color: #9E9E9E;
}