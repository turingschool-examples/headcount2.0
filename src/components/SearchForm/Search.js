import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class Search extends Component {
  constructor() {
    super();
    this.state = {
      query: ''
    };
  }

  handleChange =  event => {
    const { value } = event.target;

    this.props.filterDistricts(value);

    this.setState({ query: value });
  }

  render() {
    const { query } = this.state;

    return (
      <input
        onChange={this.handleChange}
        type="text"
        name="query"
        value={query}
        placeholder="Search"
      />
    );
  }
}

Search.propTypes = {
  filterDistricts: PropTypes.func.isRequired
};

export default Search;
