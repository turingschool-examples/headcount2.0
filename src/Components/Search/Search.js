import React from 'react';
import PropTypes from 'prop-types';
import './Search.css';

const Search = ( {handleSearch} ) => {
  return (
      <div className="search-container">
        <input 
        placeholder="Search for district..."
        className="search"
        onChange={ (e) => handleSearch(e.target.value) }/>
      </div>
    )
}

Search.propTypes = {
  handleSearch: PropTypes.func.isRequired
}

export default Search;