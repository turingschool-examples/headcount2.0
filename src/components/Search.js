import React, { Component } from 'react';
import DistrictRepository from '../helper';
import '../styles/Search.css';

class Search extends Component {
  constructor(props){
		super(props);
    this.state=  {
      userInput: ""
    };
	}

	handleChange =  (event) => {
		this.setState( {
			userInput: event.target.value
		});
	}

	handleSubmit = (event) => {
		event.preventDefault();
		this.props.setLocationData(this.state);
		this.state.userInput = "";
	}
	
	render() {
		return (
			<form onSubmit={this.handleSubmit}>
				<input
          type= "text"
					onChange= {this.handleChange}
					placeholder="Search Districts"
					value= {this.state.userInput}
					className="search-input" 
				/>
				<button className="search-button">Submit</button> 
			</form>
		);
	}
}


export default Search;