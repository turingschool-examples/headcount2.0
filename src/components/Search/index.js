import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './Search.css'

class Search extends Component {
  constructor({findAllMatches}) {
    super()
    this.state = {
      query: ''
    }
  }

  handleInputChange = event => {
    const { value } = event.target

    this.setState({query: value})
    this.props.findAllMatches(value);
  }

  handleSubmit = event => {
    event.preventDefault()
  }

  render() {
    return(
      <form className="Search" onSubmit={this.handleSubmit} >
        <input
          className="search-input"
          aria-label="Search for a specific district here"
          placeholder="Search for a specific district"
          name="query"
          value={this.state.query}
          onChange={this.handleInputChange}
        />
      </form>
    )
  }
}

Search.propTypes = {
  findAllMatches: PropTypes.func.isRequired
}

export default Search