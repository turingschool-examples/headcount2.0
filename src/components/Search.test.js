import React from 'react';
import ReactDOM from 'react-dom';
import Search from './Search';
import {shallow, mount} from 'enzyme';

describe('Search', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should have an initial state of user input', () => {
    const wrapper = shallow(<Search />);

    expect(wrapper.state().userInput).toEqual("");
  });

  it('should update the state when user inputs a string', () => {
    const wrapper = mount(<Search />)
    const event = {target: {value: 'abc'}};
    const expectedState = {
      userInput: 'abc'
    };

    wrapper.instance().handleChange(event);
    expect(wrapper.state()).toEqual(expectedState);
  });

  it('calls the setLocation prop on submit and sets the userInput back to an empty string', () => {
    
    const mockEvent = {preventDefault: () => {}}

    const wrapper = shallow(<Search setLocationData={jest.fn()} />)
    const expectedState = {
      userInput: ""
    };

    wrapper.instance().handleSubmit(mockEvent);
    expect(wrapper.state()).toEqual(expectedState);

  }); 
});
