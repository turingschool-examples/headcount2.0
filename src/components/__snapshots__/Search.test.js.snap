// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search Districts"
          type="text"
          value=""
/>,
        <button
          className="search-button"
>
          Submit
</button>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "onChange": [Function],
          "placeholder": "Search Districts",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "search-button",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search Districts"
            type="text"
            value=""
/>,
          <button
            className="search-button"
>
            Submit
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Search Districts",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "search-button",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
